tests/unit/store/waveform.spec.js

```diff
@@ -209,19 +215,39 @@ describe("store/waveform", () => {
       beforeEach(async () => {
         mocked_axios
           .onGet(get_available_data_regex) // We pass in_simulation_mode true and validate default false is replaced
-          .reply(200, nr);
+          .reply(200, nr)
+          .onGet(system_status_when_calibrating_regexp)
+          .reply(400);
 
         await store.dispatch("waveform/start_get_waveform_pinging");
       });
       test("When start_get_available_data is dispatched, the waveform_ping_interval_id does not change and setInterval is not called again", async () => {
         const spied_set_interval = jest.spyOn(window, "setInterval");
-        const initial_interval_id = store.state.waveform_ping_interval_id;
+        const initial_interval_id =
+          store.state.waveform.waveform_ping_interval_id;
         await store.dispatch("waveform/start_get_waveform_pinging");
 
-        expect(store.state.waveform_ping_interval_id).toEqual(
+        expect(store.state.waveform.waveform_ping_interval_id).toEqual(
           initial_interval_id
         );
       });
+      test("When start_get_available_data is under progress an systems_status request returns 404, then the interval is cleared and the waveform_ping_interval_id becomes null", async () => {
+        const spied_set_interval = jest.spyOn(window, "setInterval");
+        const initial_interval_id =
+          store.state.waveform.waveform_ping_interval_id;
+
+        expect(initial_interval_id).toBeGreaterThanOrEqual(0);
+
+        await store.dispatch("playback/start_calibration");
```

Comment:
Javascript is executed in an environment of `single threaded system` even if the interval timer is expired and function call back happens, it is not going to be executed until a previous function completion to that of `waveform_ping_interval_id`, looks like still the foundational working of Chrome V8 is not understood. 

The sequence of start_calibration starts the pinging function only after the prior timer of the `waveform_ping_interval_id `function is completed, and intermediate functions of the testcases are executed and  until the line 242 `await store.dispatch("playback/start_calibration");`  till then the system doesn't understand or knows that `status_ping_interval_id` is having a function ID associated. 
The dispatch results in the callback function `start_status_pinging` associated to the dispatch invoked and as prior determined even before the value `status_ping_interval_id` is committed to the store at a later stage.   In the context of the execution there is an assignment to be updated for the Vuex Store via mutation exists, but its only queued after the present function of `ping_system_status` is executed at timer expiry of 1000 millis which in `flask/actions.js function start_status_pinging` at line 75 prior to the next statement `context.commit("set_status_ping_interval_id", new_interval_id")`.   So in between above below of the dispatch start_calibration, the value of status_ping_interval_id would obviously be null because its waiting in 
queue to update in the Vuex.

The ERROR handling 404 happens at line 21 of `await call_axios_get_from_vuex(...)`which clears the value of status_ping_interval_id.

So, the statement like `begin is a side effect` is a kind of imagination of non Javascript working thinking were every timer runs on as a separate thread in a other engines which is not available for Javascript.

There is no problem with the present working and testing of the ERROR handling.
Nothing needs a change.